mllib/src/test/scala/org/apache/spark/ml/regression/GBTRegressorSuite.scala

```diff
@@ -296,6 +304,35 @@ class GBTRegressorSuite extends MLTest with DefaultReadWriteTest {
     }
   }
 
+  test("training with sample weights") {
+    val df = linearRegressionData
+    val numClasses = 0
+    // (maxIter, maxDepth)
+    val testParams = Seq(
+      (5, 5),
+      (5, 10)
+    )
+
+    for ((maxIter, maxDepth) <- testParams) {
+      val estimator = new GBTRegressor()
+        .setMaxIter(maxIter)
+        .setMaxDepth(maxDepth)
+        .setSeed(seed)
+        .setMinWeightFractionPerNode(0.1)
+
+      MLTestingUtils.testArbitrarilyScaledWeights[GBTRegressionModel,
+        GBTRegressor](df.as[LabeledPoint], estimator,
+        MLTestingUtils.modelPredictionEquals(df, _ ~= _ relTol 0.1, 0.95))
```

Comment:
Compared to `DecisionTreeRegressorSuite`, I need to limit the number of trees and loose the tolerance eps(0.99 -> 0.95) to pass the cases.
I wonder if it is due to accumulated errors among trees.

Reply:
interesting, will need to take a closer look...